front_end/src/components/searchResultOverview/FiltersWindow.js

```diff
@@ -0,0 +1,346 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+
+// Additional Libraries Imports
+import ReactStars from "react-stars";
+import NumberFormat from "react-number-format";
+
+// Material UI Imports
+import {
+  withStyles,
+  withWidth,
+  Checkbox,
+  FormControlLabel,
+  Grid,
+  Divider,
+  Typography,
+  TextField,
+  ExpansionPanel,
+  ExpansionPanelSummary,
+  ExpansionPanelDetails,
+  MenuItem,
+  Button,
+  Menu
+} from "@material-ui/core";
+import { isWidthDown } from "@material-ui/core/withWidth";
+import { ExpandMore, FilterList } from "@material-ui/icons";
+
+// Component styling
+let styles = theme => ({
+  rating: {
+    padding: 0,
+    margin: 0
+  },
+  subtitles: { fontWeight: "bold", color: "#808080" },
+  pad10: {
+    padding: 10
+  }
+});
+
+// Custom Number Format Component for Price Range TextFields
+// Ensures that user input is restricted to non-negative integers
+let NumberFormatCustom = props => {
+  let { inputRef, onChange, ...other } = props;
+  return (
+    <NumberFormat
+      {...other}
+      getInputRef={inputRef}
+      onValueChange={values => {
+        onChange({
+          target: {
+            value: values.value
+          }
+        });
+      }}
+      allowNegative={false}
+      thousandSeparator
+      prefix="$"
+    />
+  );
+};
+
+// React Functional Component (has no states) for the filters window of SearchResultOverview component
+class FiltersWindow extends Component {
+  constructor() {
+    super();
+
+    this.state = {
+      priceRangeAnchorEl: null,
+      priceRangeEquality: "To"
+    };
+
+    this.handleEqualityMenuClose = this.handleEqualityMenuClose.bind(this);
+    this.handleEqualityMenuOpen = this.handleEqualityMenuOpen.bind(this);
+  }
+
+  handleEqualityMenuOpen = event => {
+    event.preventDefault();
+    this.setState({
+      priceRangeAnchorEl: event.currentTarget
+    });
+  };
+
+  handleEqualityMenuClose = value => event => {
+    event.preventDefault();
+    this.setState({
+      priceRangeEquality: value,
+      priceRangeAnchorEl: null
+    });
+  };
+
+  render() {
+    let { priceRangeEquality, priceRangeAnchorEl } = this.state;
+
+    let {
```

Comment:
it is recommended to use **const** instead of **var, let** unless you have to

Reply:
Using 'let' limits the scope of the variable to render() {} while const sets the scope to the entire file.